apigen-deps/src/main/java/com/distelli/graphql/ResolverDataFetcher.java

```diff
@@ -37,16 +50,19 @@ public ResolverDataFetcher(DataFetcher fetcher, Resolver resolver, int listDepth
     public Object get(DataFetchingEnvironment env) {
         List<Object> unresolved = new ArrayList<>();
         Object result;
-        int depth = listDepth;
+        final int depth = listDepth;
         if ( env.getSource() instanceof List ) { // batched.
             result = getBatched(env);
-            if ( null != resolver ) addUnresolved(unresolved, result, ++depth);
+            if (nonNull(resolver) || nonNull(asyncResolver)) addUnresolved(unresolved, result, depth + 1);
```

Comment:
We still need to increment depth, otherwise replaceResolved() will not work correctly when the field is a List of objects to be resolved.